File: src/main/java/net/wanhack/utils/exp/TokenType.java
/*
 *  Copyright 2005 The Wanhack Team
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package net.wanhack.utils.exp;

public enum TokenType {
    NUMBER("number"), 
    IDENTIFIER("identifier"),
    LPAR("("),
    RPAR(")"),
    PLUS("+"),
    MINUS("-"),
    MUL("*"),
    DIV("/"),
    MOD("%"),
    COMMA(","),
    EOF("eof");
    
    private final String s;

    private TokenType(String s) {
        this.s = s;
    }
    
    @Override
    public String toString() {
        return s;
    }
}

File: core/src/main/kotlin/dev/komu/kraken/utils/exp/TokenType.kt
package dev.komu.kraken.utils.exp

enum class TokenType(val s: String) {
    NUMBER("number"),
    IDENTIFIER("identifier"),
    LPAR("("),
    RPAR(")"),
    PLUS("+"),
    MINUS("-"),
    MUL("*"),
    DIV("/"),
    MOD("%"),
    COMMA(","),
    EOF("eof");

    override fun toString() = s
}
